docs/PowerStack.md

```diff
@@ -2,47 +2,123 @@
 
-Explaination and list of pinout
+<img src="images/amdc-power-stack-highlighted.jpg" width="75%" />
 
-Picture of all 8 power stack connectors with arrows which correspond to firmware names etc
+<img src="images/amdc-input-screw-terminals.jpg" width="75%" />
 
-## Schematics
+### 1. PWM Signals
 
-Discussion related to schematics for power stack interface...
+The PWM gate drive signals are generated by the PicoZed FPGA at 1.8V. These are then level-shifted up to the user-supplied `VDRIVE` voltage. Each PWM signal then goes to the DB-15 connector. Each PWM signal is ANDed with the E-STOP signal. Thus, when the E-STOP is opened, the PWM signals go to a logic low state.
 
-## PCB Layout
+### 2. Status Signals
+
+Each inverter has four status lines (labeled A, B, C, and D). These signals operate at the `VDD` voltage level, which is configurable by the jumper on the PCB. Each status line (A/B/C/D) can be configured as either an input or output, depending on the jumper selection. Note that all eight inverters use the same directionality for the status signals (i.e. if status signal A is an input, it is an input for all eight inverters).
+
+These status signals can be used as arbitrary I/O between the AMDC and the power stack. Typically, they are used to read faults states, reset the power stack, read ready states from the power stack gate drives, etc. The voltage level which is selected by the jumper on the AMDC is supplied to each DB-15 connector so the power stack can use the correct I/O voltage level for communication.
+
+#### Jumpers
+
+On the AMDC, the following are the jumpers used to set status line voltage level and direction:
+
+- JP1: Status voltage level
+- JP4: Status A direction
+- JP5: Status B direction
+- JP6: Status C direction
+- JP7: Status D direction
+
+### 3. Voltage Rails
+
+There are three voltage rails which must be understood: `VDD`, `VDRIVE`, and `VPS`.
+
+#### `VPS` &mdash; *Power Stack General Voltage Supply*
+
+This voltage rail (the "power stack voltage rail") is completely independent from the AMDC; the AMDC simply acts as a power distribution network for this rail. On the AMDC, one screw terminal input is used to supply both the + and - rail of `VPS`. Then, the AMDC routing distributes this power rail to each DB-15 connector. This power rail is designed to supply the bulk power required by the power stack gate drivers. Usually, 12V or 24V is used for this. Note that the AMDC trace routing for this signal can only support 2A. At 24V, this would mean 48W power consumption for all power stack gate drives (or 6W each). If the user needs a more powerful gate drive, they must use an external power source for it.
+
+Note that this voltage rail passes through a 1206 0R resistor for each DB-15 connector (i.e. each inverter). Thus, the user may depopulate this resistor if needed to remove this voltage rail from the connector.
+
+#### `VDRIVE` &mdash; *Power Stack PWM Voltage Level*
```

Comment:
Can we provide an estimate of the required power supply rating to support the AMDC PWM drivers (assuming that the power stack is not drawing current)? ... trying to make it easy for a user to know how to size this external supply. This can be a crude answer, i.e., "the Vdrive supply must be able to provide at least 200 mA to support all PWM circuitry running at 1 MHz" (if that is true).

Reply:
@elsevers how's this: https://github.com/Severson-Group/AMDC-Hardware/commit/d03b9a5980034596ec67b599fdbc3e948a81f306

Reply:
I am confused by your calculation --

1. Energy to charge a capacitor = 1/2 C V<sup>2</sup>. So, we can get power as 1/2 f<sub>sw</sub>C V<sup>2</sup>. It seems that we are missing the 1/2?

2. Do you mean this equation to be per PWM channel? Or are you assuming a capacitance value that corresponds to cumulative value of all PWM channels? (i.e., you are assuming each channel has 10nF/(8 inv * 6 PWM) = 0.2 nF. Do you have reason to estimate it at this value?

Reply:
@elsevers I have used the equation from the [datasheet](http://ww1.microchip.com/downloads/en/DeviceDoc/21425C.pdf) for this. I believe they remove the 1/2 due to full cycle being both charging and discharging. Then, just multiply by Fsw.

For the capacitance value, I am assuming lumped sum for all 48x PWM outputs. Note that this changes depending on how many inverters the user is using. I have no idea what this value is to be honest. I picked 10nF since I thought 100s of pF seemed reasonable for on PCB traces... So 10 to 100x larger for off board cables kind of made sense. If you have a better way to estimate C, that would be great.

I think in general, the power requirements for VDRIVE supply are small. Unless you switch in the many MHz, with all inverters, at high VDRIVE voltages... Nominally, people don't do this.

Reply:
Thanks. I looked at the datasheet and these numbers agree well with the supply curves and the nominal power under electrical characteristics that they provide. I was surprised to see such a low supply current, given how much output current this part is capable of providing. But this seems to be right.